File: src/main/java/kitchenpos/eatinorders/tobe/domain/OrderLineItems.java
package kitchenpos.eatinorders.tobe.domain;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Embeddable;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;

import java.util.List;

@Embeddable
public class OrderLineItems {
    @OneToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinColumn(
            name = "order_id",
            nullable = false,
            columnDefinition = "binary(16)",
            foreignKey = @ForeignKey(name = "fk_order_line_item_to_orders")
    )
    private List<EatInOrderLineItem> orderLineItems;


    protected OrderLineItems() {
    }

    private OrderLineItems(List<EatInOrderLineItem> orderLineItems) {
        this.orderLineItems = orderLineItems;
    }

    public static OrderLineItems of(List<EatInOrderLineItem> orderLineItems) {
        return new OrderLineItems(orderLineItems);
    }

    public Integer size() {
        return orderLineItems.size();
    }
}

File: src/main/java/kitchenpos/eatinorders/tobe/domain/OrderLineItems.kt
package kitchenpos.eatinorders.tobe.domain

import jakarta.persistence.CollectionTable
import jakarta.persistence.ElementCollection
import jakarta.persistence.Embeddable
import jakarta.persistence.JoinColumn

@Embeddable
class OrderLineItems(
    @ElementCollection
    @CollectionTable(
        name = "order_line_items", joinColumns = [
            JoinColumn(name = "order_id")
        ]
    )
    val orderLineItemList: List<OrderLineItem>,
) {
    init {
        require(orderLineItemList.isNotEmpty()) { "주문 상품은 1개 이상이어야 합니다." }
    }
}
